src/main/java/com/microsoft/sqlserver/jdbc/SQLCollation.java

```diff
@@ -602,7 +600,23 @@ final Encoding checkSupported() throws UnsupportedEncodingException
         return this;
     }
 
-    final String charsetName() { return charsetName; }
+    final Charset charset() throws SQLServerException
+    {
+        try
+        {
+            checkSupported();
+            if (charset == null)
+            {
+                charset = Charset.forName(charsetName);
```

Comment:
@marschall thanks for the very detailed explanation, now I understand why you have introduced a new member variable here. I wanted the code to be consistent with the current implementation, but your cache implementation is better than the current. I am going to accept the PR once the testing is done. thanks again.